src/main/cpp/DragonVision/DragonCamera.h

```diff
@@ -0,0 +1,118 @@
+        OFF,
+        MACHINE_LEARNING,
+        APIRL_TAG,
+        COLOR_THRESHOLD
+    };
+
+    DragonCamera(
+        std::string cameraName,                /// <I> camera name/type
+        PIPELINE pipeline,                     /// <I> enum for pipeline
+        units::length::inch_t mountingXOffset, /// <I> x offset of cam from robot center (forward relative to robot is positive)
+        units::length::inch_t mountingYOffset, /// <I> y offset of cam from robot center (left relative to robot is positive)
+        units::length::inch_t mountingZOffset, /// <I> z offset of cam from robot center (up relative to robot is positive)
+        units::angle::degree_t pitch,          /// <I> - Pitch of camera
+        units::angle::degree_t yaw,            /// <I> - Yaw of camera
+        units::angle::degree_t roll            /// <I> - Roll of camera
+    );
+    DragonCamera() = delete;
+
+    virtual bool HasTarget() const = 0;
+
+    // Getters
+    virtual units::angle::degree_t GetTargetHorizontalOffset() const = 0;
+    virtual units::angle::degree_t GetTargetHorizontalOffsetRobotFrame(units::length::inch_t *targetDistOffset_RF, units::length::inch_t *targetDistfromRobot_RF) const = 0;
+    virtual units::angle::degree_t GetTargetVerticalOffsetRobotFrame(units::length::inch_t *targetDistOffset_RF, units::length::inch_t *targetDistfromRobot_RF) const = 0;
+    virtual units::angle::degree_t GetTargetVerticalOffset() const = 0;
+    virtual units::time::microsecond_t GetPipelineLatency() const = 0;
+    virtual units::angle::degree_t GetTargetSkew() const = 0;
+    virtual double GetTargetArea() const = 0;
+    virtual int GetAprilTagID() const = 0;
+
+    virtual VisionPose GetFieldPosition() const = 0;
+    virtual VisionPose GetFieldPosition(frc::DriverStation::Alliance alliance) const = 0;
+
+    //  Estimating distance
+    virtual units::length::inch_t EstimateTargetXDistance() const = 0;
+    virtual units::length::inch_t EstimateTargetYDistance() const = 0;
+    virtual units::length::inch_t EstimateTargetZDistance() const = 0;
+
+    virtual units::length::inch_t EstimateTargetXDistance_RelToRobotCoords() const = 0;
+    virtual units::length::inch_t EstimateTargetYDistance_RelToRobotCoords() const = 0;
+    virtual units::length::inch_t EstimateTargetZDistance_RelToRobotCoords() const = 0;
+
+    // Getters
+    PIPELINE GetPipeline() const { return m_pipeline; }
+    units::angle::degree_t GetCameraPitch() const { return m_pitch; }
+    units::angle::degree_t GetCameraYaw() const { return m_yaw; }
+    units::angle::degree_t GetCameraRoll() const { return m_roll; }
+    units::length::inch_t GetMountingXOffset() const { return m_mountingXOffset; }
+    units::length::inch_t GetMountingYOffset() const { return m_mountingYOffset; }
+    units::length::inch_t GetMountingZOffset() const { return m_mountingZOffset; }
+
+    // Setters
+    void SetPipeline(PIPELINE pipeline) { m_pipeline = pipeline; }
+
+    void SetCameraPosition(
```

Comment:
Do we intend to move the camera during runtime?

Reply:
Not this year, but the function is just a setter for all the position variables so it doesn't need to be maintained.